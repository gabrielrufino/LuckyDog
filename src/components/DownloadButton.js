import Icon from 'react-native-vector-icons/FontAwesome5';
import React from 'react';
import {TouchableOpacity} from 'react-native';

export default function DownloadButton({onPress}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <Icon name="download" size={30} color="#CACAAA" />
    </TouchableOpacity>
  );
}
