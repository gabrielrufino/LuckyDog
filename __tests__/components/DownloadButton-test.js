import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DownloadButton from '../../src/components/DownloadButton';

describe('components/DownloadButton.js', () => {
  it('Should renders the download button correctly', () => {
    const onPress = jest.fn();

    const tree = renderer.create(<DownloadButton onPress={onPress} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
