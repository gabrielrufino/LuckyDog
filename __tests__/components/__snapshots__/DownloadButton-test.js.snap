// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DownloadButton.js Should renders the download button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#CACAAA",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "900",
        },
      ]
    }
  >
    ï€™
  </Text>
</View>
`;
